samples/rigid_body.py

```diff
@@ -0,0 +1,106 @@
+from __future__ import print_function
+
+import espressomd
+from espressomd import thermostat
+from espressomd import integrate
+from espressomd.virtual_sites import VirtualSitesRelative
+from espressomd.io.writer import h5md
+
+import numpy as np
+
+system = espressomd.System(box_l=[1.0, 1.0, 1.0])
+system.set_random_state_PRNG()
+system.seed = system.cell_system.get_state()['n_nodes'] * [1234]
+
+sim_ID="rigid"
+h5md_file_name = "rigid.h5"
+
+system.time_step = 0.01
+skin=10.0
+system.cell_system.skin = skin
+box_l=100
+system.box_l = [box_l, box_l, box_l]
+system.thermostat.set_langevin(kT=1.0, gamma=1.0)
+
+### Particle types
+type_centre = 0
+type_A = 1
+
+
+#############################################################
+print("** Placing particles")
+#############################################################
+
+
+# start at p_id=1, and reserve p_id=0 for the centre bead.
+p_id=1
+
+branch_len=5
+x=box_l*0.5
+y=box_l*0.5
+z=box_l*0.5
+
+# place six branches, pointing +/-x +/-y and +/-z
+for n in range(branch_len):
+    system.part.add(pos=[x+(n+1), y, z], type=type_A, virtual=1)
+for n in range(branch_len):
+    system.part.add(pos=[x-(n+1), y, z], type=type_A, virtual=1)
+for n in range(branch_len):
+    system.part.add(pos=[x, y+(n+1), z], type=type_A, virtual=1)
+for n in range(branch_len):
+    system.part.add(pos=[x, y-(n+1), z], type=type_A, virtual=1)
+for n in range(branch_len):
+    system.part.add(pos=[x, y, z+(n+1)], type=type_A, virtual=1)
+for n in range(branch_len):
+    system.part.add(pos=[x, y, z-(n+1)], type=type_A, virtual=1)
+
+
+#############################################################
+# rigid stuff
+#############################################################
+system.virtual_sites= VirtualSitesRelative(have_velocity=True)
+
+#here we calculate the center of mass position (com) and the moment of inertia (momI) of the object
+com = system.analysis.center_of_mass(part_type=type_A)
+print("center of mass is:", com)
+
+# only change this for multiple GPUS
```

Comment:
you can just leave the statement in and check for the number of nodes (not GPUs)